generators/git/index.js

```diff
@@ -22,13 +28,34 @@ module.exports = generators.Base.extend({
       this.templatePath('gitignore'),
       this.destinationPath('.gitignore')
     );
+
+    var done = this.async();
+
+    originUrl(this.destinationPath(), function (err, url) {
+      this.originUrl = url;
+      done();
+    }.bind(this));
+
+  },
+
+  writing: function () {
+    this.pkg = this.fs.readJSON(this.destinationPath('package.json'), {});
+    if (!this.pkg.repository) {
+      if (this.originUrl) {
+        this.pkg.repository = this.originUrl;
+      } else {
+        this.pkg.repository = this.options.githubAccount + '/' + this.options.name;
+      }
+    }
+
+    this.fs.writeJSON('package.json', this.pkg);
   },
 
   end: function () {
     this.spawnCommandSync('git', ['init']);
 
-    if (!this.options.repositoryPath) {
-      var repoSSH = 'git@github.com:' + this.options.repositoryPath + '.git';
+    if (!this.originUrl) {
+      var repoSSH = 'git@github.com:' + this.pkg.repository + '.git';
```

Comment:
At this point, it feels to me like we have no garantee `this.pkg.repository` is a `username/repo` string. It could literally be anything; very likely a full url to a git repository.